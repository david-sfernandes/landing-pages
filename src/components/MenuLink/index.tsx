import Container from "./style";

export default function MenuLink({children, link, newTab = false}: MenuLinkProps) {
  const target = newTab ? '_blank' : '_self'
  return (
    <Container href={link} target={target}>
      {children}
    </Container>
  )
}