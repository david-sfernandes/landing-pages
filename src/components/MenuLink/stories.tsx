import MenuLink from ".";

const story = {
  title: "MenuLink",
  component: MenuLink,
  args: {
    children: "MenuLink",
    link: 'https://www.google.com.br'
  },
  argTypes: {
    children: { type: "string" }
  }
};
export default story;

export const Template = (args: any) => (
  <div style={{ display: 'flex'}}>
    <MenuLink {...args} />
  </div>
);
