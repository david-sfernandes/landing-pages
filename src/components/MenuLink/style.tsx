import styled, { css } from "styled-components";

const Container = styled.a`
  ${({theme}) => css`
    font-size: ${theme.font.sizes.sm};
    display: block;
    text-decoration: none;
    padding: ${theme.spacings.sm};
    color: ${theme.colors.primary};
    position: relative;

    &::after {
      content: '';
      position: absolute;
      bottom: 7px;
      left: 25%;
      width: 0;
      height: 3px;
      background: ${theme.colors.secundary};
      transition: all 300ms ease-in-out;
    }

    &:hover::after {
      left: 25%;
      width: 50%;
    }
  `}
`

export default Container;