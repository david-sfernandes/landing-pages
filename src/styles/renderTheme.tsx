import { render } from "@testing-library/react";
import { ReactNode } from "react";
import { ThemeProvider } from 'styled-components';
import { theme } from "./theme";

export default function renderTheme(children: ReactNode) {
  return (
    render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)
  )
}