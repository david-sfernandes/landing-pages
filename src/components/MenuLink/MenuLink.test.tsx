import { screen } from "@testing-library/react";
import MenuLink from ".";
import renderTheme from "../../styles/renderTheme";

describe("<MenuLink />", () => {
  it("should render a link", () => {
    renderTheme(<MenuLink link='http://localhost'>Text</MenuLink>);

    expect(screen.getByRole('link', {name: 'Text'})).toHaveAttribute('target', '_self');
  });

  it("should open in a new tab", () => {
    renderTheme(<MenuLink link='http://localhost' newTab>Text</MenuLink>);
  
    expect(screen.getByRole('link', {name: 'Text'})).toHaveAttribute('target', '_blank');
  });

  it("should match snapshot", () => {
    renderTheme(<MenuLink link='http://localhost' newTab>Text</MenuLink>);
  
    expect(screen.getByRole('link', {name: 'Text'})).toMatchSnapshot();
  });
});