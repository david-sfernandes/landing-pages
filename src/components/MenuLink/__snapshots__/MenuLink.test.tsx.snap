// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuLink /> should match snapshot 1`] = `
.c0 {
  font-size: 16px;
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 16px;
  color: #0A1128;
  position: relative;
}

.c0::after {
  content: '';
  position: absolute;
  bottom: 7px;
  left: 25%;
  width: 0;
  height: 3px;
  background: #dc143c;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0:hover::after {
  left: 25%;
  width: 50%;
}

<a
  class="c0"
  href="http://localhost"
  target="_blank"
>
  Text
</a>
`;
